File: day2_part1_2015.fs

module Day2

let input = System.IO.File.ReadAllLines "input.txt"

let calculateSurfaceArea l w h =
    let sides = [l*w; w*h; h*l]
    let minSide = List.min sides
    let totalArea = 2*l*w + 2*w*h + 2*h*l
    totalArea + minSide

let totalSquareFeet = 
    input 
    |> Array.map (fun line -> 
        let dimensions = line.Split 'x' |> Array.map int
        let l = dimensions.[0]
        let w = dimensions.[1]
        let h = dimensions.[2]
        calculateSurfaceArea l w h)
    |> Array.sum

printfn "%d" totalSquareFeet

File: erlang/day2_part1_2015.erl
-module(task).
-export([call/0]).

call() ->
    {ok, File} = file:read_file("input.txt"),
    Lines = string:split(binary_to_list(File), "\n", all),
    TotalArea = lists:foldl(fun(Present, Acc) ->
        [L, W, H] = string:tokens(Present, "x"),
        LNum = list_to_integer(L),
        WNum = list_to_integer(W),
        HNum = list_to_integer(H),
        Side1 = LNum * WNum,
        Side2 = WNum * HNum,
        Side3 = HNum * LNum,
        MinSide = lists:min([Side1, Side2, Side3]),
        2*LNum*WNum + 2*WNum*HNum + 2*HNum*LNum + MinSide + Acc
    end, 0, Lines),
    io:format("~p~n", [TotalArea]).